5-VM-Ubuntu-1-NodeTypes-Secure/azuredeploy.json

```diff
@@ -1,39 +1,40 @@
 {
-  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json",
+  "$schema": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json",
   "contentVersion": "1.0.0.0",
   "parameters": {
-    "location": {
+    "clusterLocation": {
       "type": "string",
-      "defaultValue": "[resourceGroup().location]",
+      "defaultValue": "westus",
       "metadata": {
         "description": "Location of the Cluster"
       }
     },
-    "clusterDnsName": {
+    "clusterName": {
       "type": "string",
-      "minLength": 3,
-      "maxLength": 23,
+      "defaultValue": "GEN-UNIQUE",
```

Comment:
GEN values are not valid in the template file - use [concat('sf-', uniqueString(resourceGroup().id))]

Reply:
Done.